### PR TITLE
Amqp async

### PR DESCRIPTION
**Pull request by vso-tc**

### CODE REVIEW

File: amqp/src/main/java/io/github/tcdl/adapters/amqp/AmqpConsumerAdapter.java [hunk 1]
```diff
@@ -69,20 +53,12 @@ public void subscribe(RawMessageHandler msgHandler) {
...
 
-            consumerTag = channel.basicConsume(queueName, false /* autoAck */, new DefaultConsumer(channel) {
-                @Override
-                public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProperties properties, byte[] body) throws IOException {
-                    getChannel().basicAck(envelope.getDeliveryTag(), false);
-                    String bodyStr = new String(body);
-                    msgHandler.onMessage(bodyStr);
-                }
-            });
+            consumerTag = channel.basicConsume(queueName, false /* autoAck */, new AmqpMsbConsumer(channel, consumerThreadPool, msgHandler));
```
Comment (vso-tc) [inline amqp/src/main/java/io/github/tcdl/adapters/amqp/AmqpConsumerAdapter.java:56]: This callback consumer has been moved to separate class

File: amqp/src/main/resources/amqp.conf [hunk 2]
```diff
@@ -5,5 +5,6 @@ config.amqp = {
   port = "5672"
   groupId = "msb-java"
   durable: false
+  consumerThreadPoolSize: 5
```
Comment (vso-tc) [inline amqp/src/main/resources/amqp.conf:8]: The new handle that allows to tweak performance.

File: amqp/src/test/resources/log4j.xml [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
     <appender name="consoleAppender" class="org.apache.log4j.ConsoleAppender">
         <layout class="org.apache.log4j.PatternLayout">
-            <param name="ConversionPattern" value="%d{dd MMM yyyy HH:mm:ss} %5p %c{1} - %m%n"/>
+            <param name="ConversionPattern" value="%d{dd MMM yyyy HH:mm:ss} %5p [%t] %c{1} - %m%n"/>
```
Comment (vso-tc) [inline amqp/src/test/resources/log4j.xml:8]: Including thread name in logger output.

File: amqp/src/main/java/io/github/tcdl/adapters/amqp/AmqpMsbConsumer.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ * will not be processed in parallel.
+ *
+ * To address this issue {@link AmqpMsbConsumer} just takes incoming message, wraps it in a task and puts into a thread pool. And the actual
+ * processing is happening in the separate thread from that thread pool.
+ */
+public class AmqpMsbConsumer extends DefaultConsumer {
+
+    private static final Logger logger = LoggerFactory.getLogger(AmqpMsbConsumer.class);
+
+    ExecutorService consumerThreadPool;
```
Comment (rdrozdov-tc) [inline amqp/src/main/java/io/github/tcdl/adapters/amqp/AmqpMsbConsumer.java:30]: do we need it with default visibility?

File: amqp/src/main/java/io/github/tcdl/adapters/amqp/AmqpMsbConsumer.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    RawMessageHandler msgHandler;
```
Comment (rdrozdov-tc) [inline amqp/src/main/java/io/github/tcdl/adapters/amqp/AmqpMsbConsumer.java:31]: the same

File: amqp/src/main/java/io/github/tcdl/adapters/amqp/AmqpMsbConsumer.java (hunk 4 above)
Comment (vso-tc) [inline amqp/src/main/java/io/github/tcdl/adapters/amqp/AmqpMsbConsumer.java:30]: Yep, to use in unit tests

File: amqp/src/main/java/io/github/tcdl/adapters/amqp/AmqpMsbConsumer.java (hunk 5 above)
Comment (vso-tc) [inline amqp/src/main/java/io/github/tcdl/adapters/amqp/AmqpMsbConsumer.java:31]: the same :)

**Pull request merged**